### DISCUSSION TITLE
Import with Merge

### DISCUSSION THREAD

**Discussion by Noisycall** [ananthakumaran/paisa#163] [Ideas]
Basically a way to add data to an existing ledger using import. Would be useful in combination with predict account as I can manually tag things once then let predict do the heavy lifting.

**Comment by ananthakumaran** [comment]
Have you looked at [include](https://paisa.fyi/reference/journal/) option. I usually import month wise and create a journal file per month. The main journal would just include the entire folder. Predict uses all the existing transactions (doesn't matter which file). The editor also handles nested folder/files well. I would usually check if my credit card statement balance matches what ledger says just by looking at the balance shown on the editor page

```
include expenses/2023/*.ledger
```

Let me know if you have different use case. Merging means I have to first define what is considered as duplicate and so forth. I have been mostly avoiding it so far, might have to do it at some point though.

**Reply by Noisycall** [reply]
Ah, I see what you are saying. Basically have a parent ledger to which all other ledgers are added. This is definitely more powerful, if a little less obvious directly (hence I asked). I can see that merging while dealing with duplicates would be difficult, maybe to start with you don't deal with that? Just let the user use the import tool smartly.

**Reply by ananthakumaran** [reply]
> maybe to start with you don't deal with that

yup, if you have a journal per month per bank/cc/account you can just overwrite the full file if you ever want to re-import.

I am a little reluctant to add merge support because Paisa supports ledger/beancount/hledger and merging all these formats without breaking stuff might not be very easy. And some of the things (like balance assertion) are very order sensitive, so re-arranging them might not be possible. Even if I support merge, there will be always corner cases, so trying to find other ways to handle them without resorting to merge.

> if a little less obvious directly (hence I asked)

I added this to the doc after a user asked about some other feature (that could be solved via include). Hopefully discussions like this would show up when the next user search for merge :)
